docs/user-guide/templates.md

```diff
@@ -168,6 +173,63 @@ jobs:
 
 ```
 
+### Step order
```

Comment:
Main heading should be `Template composition/inheritance` `Order` then being a sub-heading under it

Reply:
I saw this as something else to go under `Overriding Template Steps`, but I can move it